### PR TITLE
Improve Metrics fo Auction Building

### PR DESCRIPTION
**Pull request by nlordell**
This PR adds a new gauge for the number of **market** orders that are filtered out because of missing prices. This is because filtering out market orders for this reason is more serious than limit orders. This new metric will allow us to monitor this separately and potentially create alerts to deal with missing market order native token prices.

I also changed the `checkpoint` to log the order's class to make debugging filtered orders in the log easier.

### Test Plan

Rust compiler

### CODE REVIEW

**Review by sunce86 — APPROVED** [review]

File: crates/autopilot/src/solvable_orders.rs [hunk 1]
```diff
@@ -45,6 +45,10 @@ pub struct Metrics {
     /// Auction filtered orders grouped by class.
     #[metric(labels("reason"))]
     auction_filtered_orders: IntGaugeVec,
+
+    /// Auction filtered market orders due to missing native token price.
+    #[metric()]
```
Comment (MartinquaXD) [inline crates/autopilot/src/solvable_orders.rs:50]: Does this line serve a purpose?

**Review by MartinquaXD — APPROVED** [review]

File: crates/autopilot/src/solvable_orders.rs [hunk 2]
```diff
@@ -485,10 +494,21 @@ fn get_orders_with_native_prices(
                 used_prices.insert(*t1, *p1);
                 true
             }
-            _ => false,
+            _ => {
+                if order.metadata.class == OrderClass::Market {
+                    filtered_market_orders += 1;
+                }
```
[the remark this answers is not in the document]
Comment (nlordell) [reply crates/autopilot/src/solvable_orders.rs:500]: Love it.

File: crates/autopilot/src/solvable_orders.rs (hunk 1 above)
Comment (nlordell) [reply crates/autopilot/src/solvable_orders.rs:50]: It appears to be needed. We also use it in other places:

https://github.com/cowprotocol/services/blob/5dcfab50fa1a5b8b3189a2d38685fa52181e2d73/crates/shared/src/maintenance.rs#L134-L146

Comment (vkgnosis) [reply crates/autopilot/src/solvable_orders.rs:50]: It's not required.

**Pull request merged**